src/Routing/Asset.php

```diff
@@ -0,0 +1,356 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Routing;
+
+use Cake\Core\Configure;
+use Cake\Core\Exception\Exception;
+use Cake\Core\Plugin;
+use Cake\Utility\Inflector;
+
+/**
+ * Class for generating asset URLs.
+ */
+class Asset
+{
+    /**
+     * Inflection type to use when inflection plugin/theme name to get directory name.
+     *
+     * @var string
+     * @psalm-var 'dasherize'|'underscore'
+     */
+    protected static $inflectionType = 'underscore';
+
+    /**
+     * Set inflection type.
+     *
+     * @param string $inflectionType Inflection type
+     * @return void
+     * @psalm-param 'dasherize'|'underscore' $inflectionType
+     */
+    public static function setInflectionType(string $inflectionType)
```

Comment:
you can set :void here as return type, right?

Reply:
yup, it's an oversight